src/main/java/de/siphalor/amecs/impl/mixin/MixinControlsOptionsScreen.java

```diff
@@ -32,11 +33,10 @@ public MixinControlsOptionsScreen(Screen screen, GameOptions gameOptions, Text t
 
 	@Inject(method = "mouseClicked", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/option/GameOptions;setKeyCode(Lnet/minecraft/client/option/KeyBinding;Lnet/minecraft/client/util/InputUtil$Key;)V"))
 	public void onClicked(double x, double y, int type, CallbackInfoReturnable<Boolean> callbackInfoReturnable) {
-		InputUtil.Key keyCode = ((IKeyBinding) focusedBinding).amecs$getKeyCode();
+		InputUtil.Key key = ((IKeyBinding) focusedBinding).amecs$getBoundKey();
 		KeyModifiers keyModifiers = ((IKeyBinding) focusedBinding).amecs$getKeyModifiers();
-		if (keyCode != InputUtil.UNKNOWN_KEY) {
-			int keyCodeCode = keyCode.getCode();
-			keyModifiers.set(KeyModifier.fromKeyCode(keyCodeCode), true);
+		if (!key.equals(InputUtil.UNKNOWN_KEY)) {
```

Comment:
Yes. But a mod MIGHT create a custom Keybinding like so:
```Java
new KeyBinding("key.MOD.NAME", InputUtil.Type.KEYSYM, GLFW.GLFW_KEY_UNKNOWN, "key.category.MOD");
```
and this would break that logic.
For probably this exact reason the vanilla code has it the same way:
```Java
   public boolean isUnbound() {
      return this.boundKey.equals(InputUtil.UNKNOWN_KEY);
   }```